### ISSUE TITLE
Players only spawn after selecting a weapon

### ISSUE THREAD

**Issue by kassibuss** [NeotokyoRebuild/neo#286]
Players still spawn before selecting a weapon, leading to bugs such as switching from recon to assault leading to assault having 13 of therm-optic camo. -Wakfu

(same description in issue  "Them-optic capacity is always correct" but i think it still needs a separate issue just incase) - StellaNova

**Comment by blaberry** [comment]
This is still an issue, all players spawn as soon as the round starts, instead of after having picked their weapon.

**Comment by AdamTadeusz** [comment]
Personally I think this is an improvement over the original where your camera remains fixed in the position you were in in the previous round, allowing you to observe the enemy team as they spawn and pick their weapons, or position the camera in your new spawn and watch for any enemy players rushing your new spawn, picking a class and weapon to suddenly appear in a spot they may have already checked to blast them from behind.

This is also problematic for game modes like tdm for example where a team with a lead can choose to delay their spawn to delay/prevent the enemy team from scoring points, as killing enemies is the only way of scoring points in such game modes.

**Comment by Agiel** [comment]
I didn't have any issues with the current implementation in the last playtest. So unless any serious bugs or exploits show up I think it's fine to keep it.

**Comment by nullsystem** [comment]
+1 I liked the current behavior of able to switch class+loadout in freezetime, and is also a required behavior for DM/TDM.

**Comment by kassibuss** [comment]
Yeah, current implementation is fine. Does it give the player the last used weapon though, like in the og?

**Comment by AdamTadeusz** [comment]
close as wontfix then?
